src/Mod/Part/App/FaceMakerBullseye.cpp

```diff
@@ -88,48 +99,85 @@ void FaceMakerBullseye::Build_Essence()
-        if (!planeFinder.Found())
+        if (!planeFinder.Found()) {
             throw Base::ValueError("Wires are not coplanar.");
+        }
         plane = GeomAdaptor_Surface(planeFinder.Surface()).Plane();
     }
 
-    //sort wires by length of diagonal of bounding box.
-    std::vector<TopoDS_Wire> wires = this->myWires;
-    std::stable_sort(wires.begin(), wires.end(), FaceMakerCheese::Wire_Compare());
+    std::vector<WireInfo> wireInfos;
+    for (const auto& w : this->myTopoWires) {
+        Bnd_Box box;
+        if (w.isNull()) {
+            continue;
+        }
+        BRepBndLib::AddOptimal(w.getShape(), box, Standard_False);
+        if (box.IsVoid()) {
+            continue;
+        }
+        wireInfos.emplace_back(w, box);
+    }
+
+    // Sort wires by length of diagonal of bounding box.
+    std::stable_sort(wireInfos.begin(), wireInfos.end());
 
-    //add wires one by one to current set of faces.
-    //We go from last to first, to make it so that outer wires come before inner wires.
-    std::vector< std::unique_ptr<FaceDriller> > faces;
-    for (int i = static_cast<int>(wires.size()) - 1; i >= 0; --i) {
-        TopoDS_Wire& w = wires[i];
+    for (int i = 0; i < (reuseInnerWire ? 2 : 1); ++i) {
+        // add wires one by one to current set of faces.
+        std::vector<std::unique_ptr<FaceDriller>> faces;
+        for (auto it = wireInfos.begin(); it != wireInfos.end();) {
 
-        //test if this wire is on any of existing faces (if yes, it's a hole;
-        // if no, it's a beginning of a new face).
-        //Since we are assuming the wires do not intersect, testing if one vertex of wire is in a face is enough.
-        gp_Pnt p = BRep_Tool::Pnt(TopoDS::Vertex(TopExp_Explorer(w, TopAbs_VERTEX).Current()));
-        FaceDriller* foundFace = nullptr;
-        for (std::unique_ptr<FaceDriller>& ff : faces) {
-            if (ff->hitTest(p)) {
-                foundFace = &(*ff);
-                break;
+            // test if this wire is on any of existing faces (if yes, it's a hole;
+            //  if no, it's a beginning of a new face).
+            FaceDriller* foundFace = nullptr;
+            bool hitted = false;
+            for (auto rit = faces.rbegin(); rit != faces.rend(); ++rit) {
+                switch ((*rit)->hitTest(it->wire)) {
+                    case FaceDriller::HitTest::Hit:
+                        foundFace = rit->get();
+                        hitted = true;
+                        break;
+                    case FaceDriller::HitTest::HitOuter:
+                        // Shape in outer wire but not on face, which means it is
+                        // within a hole. So it's a hit and we shall make a new face
+                        // with the wire.
+                        hitted = true;
+                        break;
+                    default:
+                        break;
+                }
```

Comment:
1. Probably can be replaced with std::find_if? But not sure. Would require returning tuple probably.
2. If not probably can replaced with `for (auto& face : faces | std::views::reverse)` - should result in code that is way easier to read.

I am however only pointing a potential better way - does not need to be addressed right now.